runner/android_junit_runner/javatests/androidx/test/internal/runner/TestRequestBuilderTest.java

```diff
@@ -227,6 +227,50 @@ public String describe() {
     }
   }
 
+  @SampleCustomAnnotationOnClass(runTests = true)
+  public static class SampleCustomFilterOnClassRunTestsTestClass {
+    @Test
+    public void testOneRun() {}
+
+    @Test
+    public void testTwoRun() {}
+  }
+
+  @SampleCustomAnnotationOnClass(runTests = false)
+  public static class SampleCustomFilterOnClassSkipTestsTestClass {
+    @Test
+    public void testOneSkip() {}
+
+    @Test
+    public void testTwoSkip() {}
+  }
+
+  public static class SampleCustomFilterOnClass extends AbstractFilter {
```

Comment:
This test code esentially simulates consumer code. The consumer shouldn't need to override anything but `evaluateTest` for this PR to work on classes. And they don't, thanks to the changes in `CustomFilters.shouldRun` above.

This class shouldn't exist. Usages can be replaced by the existing `SampleCustomFilter` class.